### PR TITLE
Follow up PR on AutoCrafters

### PR DESCRIPTION
**Pull request by svr333**
## Description
Cleans up the small remarks regarding Cookie's autocrafters PR

## Changes
- Add Validate.notNulls
- Whitelines
- Add missing access specifier to class

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java [hunk 1]
```diff
@@ -416,21 +417,20 @@ public void register(@Nonnull SlimefunAddon addon) {
         if (getCapacity() <= 0) {
             warn("The capacity has not been configured correctly. The Item was disabled.");
             warn("Make sure to call '" + getClass().getSimpleName() + "#setEnergyCapacity(...)' before registering!");
+            return;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java:420]: No

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java [hunk 2]
```diff
@@ -416,21 +417,20 @@ public void register(@Nonnull SlimefunAddon addon) {
...
         }
 
         if (getEnergyConsumption() <= 0) {
             warn("The energy consumption has not been configured correctly. The Item was disabled.");
             warn("Make sure to call '" + getClass().getSimpleName() + "#setEnergyConsumption(...)' before registering!");
+            return;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java:426]: No

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java [hunk 3]
```diff
@@ -416,21 +417,20 @@ public void register(@Nonnull SlimefunAddon addon) {
...
         }
-
-        if (getCapacity() > 0 && getEnergyConsumption() > 0) {
-            super.register(addon);
-        }
+        
+        super.register(addon);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java:429]: :NotLikeThis: Revert these changes please

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/SlimefunItemRecipe.java [hunk 4]
```diff
@@ -25,7 +26,7 @@
  * @see SlimefunAutoCrafter
  *
  */
-class SlimefunItemRecipe extends AbstractRecipe {
+public class SlimefunItemRecipe extends AbstractRecipe {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/SlimefunItemRecipe.java:29]: Don't make this public...

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java (hunk 1 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java:420]: but why? now you have two ifs checking the same thing?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/auto_crafters/AbstractAutoCrafter.java:420]: We want to send a warning if one value was misconfigured.
But we still want to send both warnings if both values were misconfigured.

That's why a `return;` here is very much unwanted since it would swallow the second warning message.
Hope that clears it up for you

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM
